pulsar-client/src/main/java/org/apache/pulsar/client/impl/PersistentAcknowledgmentsGroupingTracker.java

```diff
@@ -66,8 +67,8 @@ public class PersistentAcknowledgmentsGroupingTracker implements Acknowledgments
 
     private final long acknowledgementGroupTimeMicros;
 
-    private volatile TimedCompletableFuture<Void> currentIndividualAckFuture;
-    private volatile TimedCompletableFuture<Void> currentCumulativeAckFuture;
+    private volatile CompletableFuture<Void> currentIndividualAckFuture;
+    private volatile CompletableFuture<Void> currentCumulativeAckFuture;
```

Comment:
why not still use `TimedCompletableFuture`? seem some code needs to cast its class.type

Reply:
With `TimedCompletableFuture`, the `Optional` method chains would be:

```java
readLock.map(__ -> currentIndividualAckFuture).orElse(/* null completed TimedCompletableFuture */);
```

We need to adding a static `completedFuture` method like:

```java
    public static <U> TimedCompletableFuture<U> completedFuture(U value) {
        TimedCompletableFuture<U> future = new TimedCompletableFuture<>();
        future.complete(value);
        return future;
    }
```

But it's a little different from `CompletableFuture#completedFuture`, which only calls the constructor with the value. So I decided to keep the current logic.

Reply:
I made another change to avoid the type cast. Since `TimedCompletableFuture` is only created and used in the private `newMessageAckCommandAndWrite`, we can change the argument from `TimedCompletableFuture` to `CompletableFuture`.

/cc @Demogorgon314 @eolivelli for this change.

Reply:
The change brings a bug so I reverted it again. @congbobo184 Would the explanation makes sense to you?

Reply:
how about adding a method in `TimedCompletableFuture`
```
    public static <T> TimedCompletableFuture<T> completedFuture(T value) {
        TimedCompletableFuture<T> future = new TimedCompletableFuture<>();
        future.complete(value);
        return future;
    }
```

Reply:
Though a null completed `TimedCompletableFuture` might not make a difference with `CompletableFuture.completedFuture(null)`, it still actually changed the behavior a little that `Consumer#acknowledgeAsync` will return a `TimedCompletableFuture` instead of `CompletableFuture`.

Someone could write (though it's only theoretically)

```java
var future = consumer.acknowledgeAsync(msgId);
if (future instanceof CompletableFuture) {
    // AckReceipt is disabled
} else {
    var timedFuture = (TimedCompletableFuture) future;
    // AckReceipt is enabled
}
```

The implementations are also a little different, see my comments:
- https://github.com/apache/pulsar/pull/18392#discussion_r1021038684
- https://github.com/apache/pulsar/pull/18392#issuecomment-1310219078

> But it's a little different from CompletableFuture#completedFuture, which only calls the constructor with the value.